functions/tab.xfce4_terminal.fish

```diff
@@ -0,0 +1,3 @@
+function tab.xfce4_terminal -a cdto cmd
+  xfce4-terminal --tab --working-directory "$cdto" -e "fish -c '$cmd; exec fish'"
```

Comment:
I'd be happy merging as-is, and dealing with further optimization later. Have you seen some of the other integrations? :)